pkgs/modules/deno/default.nix

```diff
@@ -0,0 +1,32 @@
+{ pkgs, lib, ... }:
+
+let
+  inherit (pkgs) deno;
+  version = lib.versions.major deno.version;
+
+  extensions = [ ".json" ".jsonc" ".js" ".jsx" ".ts" ".tsx" ];
+in
+
+{
+  id = "deno-${version}";
+  name = "Deno Tools";
+
+  replit.packages = [
+    deno
+  ];
+
+  replit.runners.deno-script-runner = {
+    name = "deno";
+    language = "javascript";
```

Comment:
Why is it JavaScript here and below?

Reply:
why wouldn't it be? we use javascript for everything that works with both js and ts...

Reply:
this `language` field is completely arbitrary and i think just a holdout from polygott. the only reason i set it here is because i'm not confident things won't work without this field

Reply:
The "language: javascript" stuff may be better served by a set, where the set of all attributes of an environment can be tested if we know we need one thing or another, matching the most specific subset that we can find.

`{javascript, deno}` being distinct from `{javascript, typescript}` but still having some commonalities depending on what we need.

This is also a problem in upm, which is largely where I'm coming from here.

Reply:
@blast-hardcheese the worst part is, this instance of "javascript" shouldn't actually be anywhere in any of our codebases. it could be `coltonrulesz\\foobar` and wouldn't make a difference. the only reason it exists is for mapping glob patterns to language servers when they're being configured via .replit. it's an antiquated pigeonhole that serves an entirely different purpose with no beneficial heuristics we can't attach via other means